cmd/scheduler/entrypoints/precheck.go

```diff
@@ -3,83 +3,61 @@ package entrypoints
 import (
 	"context"
 	"fmt"
-	"time"
 
+	"github.com/flyteorg/flyteidl/clients/go/admin"
 	"github.com/flyteorg/flytestdlib/logger"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/metadata"
-	"google.golang.org/grpc/status"
-
-	"github.com/avast/retry-go"
-	adminClient "github.com/flyteorg/flyteidl/clients/go/admin"
-	"github.com/pkg/errors"
-	healthpb "google.golang.org/grpc/health/grpc_health_v1"
 
 	"github.com/spf13/cobra"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
 )
 
 const (
-	timeout            = 30 * time.Second
-	timeoutError       = "timeout: failed to connect service %q within %v"
-	connectionError    = "error: failed to connect service at %q: %+v"
-	deadlineError      = "timeout: health rpc did not complete within %v"
-	healthCheckError   = "Health check failed with status %v"
 	healthCheckSuccess = "Health check passed, Flyteadmin is up and running"
+	healthCheckError   = "Health check failed with status %v"
 )
 
 var preCheckRunCmd = &cobra.Command{
 	Use:   "precheck",
 	Short: "This command will check pre requirement for scheduler",
 	RunE: func(cmd *cobra.Command, args []string) error {
-		opts := []grpc.DialOption{
-			grpc.WithUserAgent("grpc_health_probe"),
-			grpc.WithBlock(),
-			grpc.WithInsecure(),
-		}
 		ctx := context.Background()
-		config := adminClient.GetConfig(ctx)
 
-		err := retry.Do(
+		// Do maximum of 30 retries on failures with constant backoff factor
+		opts := wait.Backoff{Duration: 3000, Factor: 2.0, Steps: 30, Jitter: 0.1}
```

Comment:
Can you move these to a config with these numbers as the default?

Reply:
Done.